src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -1286,7 +1403,240 @@ static int _xccdf_policy_generate_fix_other(struct oscap_list *rules_to_fix, str
+	while (oscap_iterator_has_more(package_remove_it)) {
+		char *package = (char *) oscap_iterator_next(package_remove_it);
+		_write_text_to_fd(output_fd, "-");
+		_write_text_to_fd(output_fd, package);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(package_remove_it);
+	_write_text_to_fd(output_fd, "%end\n");
+	_write_text_to_fd(output_fd, "\n");
+	return 0;
+}
+
+static int _generate_kickstart_post(struct kickstart_commands *cmds, const char *profile_id, const char *input_path, int output_fd)
+{
+	_write_text_to_fd(output_fd, "%post\n");
+	char *dup = strdup(input_path);
+	char *basename = oscap_basename(dup);
+	free(dup);
+	char *oscap_command = oscap_sprintf(
+		"oscap xccdf eval --remediate --profile '%s' /usr/share/xml/scap/ssg/content/%s\n",
+		profile_id, basename);
+	free(basename);
+	_write_text_to_fd(output_fd, "# Perform OpenSCAP hardening\n");
+	_write_text_to_fd_and_free(output_fd, oscap_command);
+	struct oscap_iterator *post_it = oscap_iterator_new(cmds->post);
+	while (oscap_iterator_has_more(post_it)) {
+		char *command = (char *) oscap_iterator_next(post_it);
+		_write_text_to_fd(output_fd, command);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(post_it);
+	_write_text_to_fd(output_fd, "%end\n");
+	_write_text_to_fd(output_fd, "\n");
+	return 0;
+}
+
+
+const char *common_partition = (
+"# Initialize (format) all disks (optional)\n"
+"zerombr\n"
+"\n"
+"# The following partition layout scheme assumes disk of size 20GB or larger\n"
+"# Modify size of partitions appropriately to reflect actual machine's hardware\n"
+"#\n"
+"# Remove Linux partitions from the system prior to creating new ones (optional)\n"
+"# --all	erase all partitions\n"
+"# --initlabel	initialize the disk label to the default based on the underlying architecture\n"
+"clearpart --all --initlabel\n"
+"reqpart\n"
+"\n"
+"# Create primary system partitions (required for installs)\n"
+"part /boot --fstype=xfs --size=512 --fsoptions=\"nodev,nosuid,noexec\"\n"
+"part pv.01 --grow --size=1\n"
+"\n"
+"# Create a Logical Volume Management (LVM) group (optional)\n"
+"volgroup VolGroup pv.01\n"
+"\n"
+"# Create particular logical volumes (optional)\n"
+"\nlogvol / --fstype=xfs --name=root --vgname=VolGroup --size=10240 --grow\n"
+);
+
+static int _generate_kickstart_logvol(struct kickstart_commands *cmds, int output_fd)
+{
+	_write_text_to_fd(output_fd, common_partition);
+	struct oscap_iterator *logvol_it = oscap_iterator_new(cmds->logvol);
+	while (oscap_iterator_has_more(logvol_it)) {
+		char *command = (char *) oscap_iterator_next(logvol_it);
+		_write_text_to_fd(output_fd, command);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(logvol_it);
+	_write_text_to_fd(output_fd, "logvol swap --name=swap --vgname=VolGroup --size=2016\n");
+	return 0;
+}
+
+const char *common_kickstart_header = (
+"# Specify installation method to use for installation\n"
```

Comment:
We should get rid of generic KS-related comments. Also, we should get rid of everything that is optional, our KS file should be the most simple thing to install the system as it would be installed by default, but hardened. Everything extra is an exercise for the advanced customer who would use the feature.

Reply:
I assume the minimal remaining stuff should still be consumable by Anaconda as-is, correct?

Or would the user need to add some things to make it work in the first place?

(I'm thinking of an analogy with osbuild and including `name`, `description`, `version`, etc.)

Reply:
The `cdrom` option should work for a default RHEL image, right?

Reply:
Depends on what you mean by "RHEL image".

I would argue it's pretty uncommon to inject `inst.ks=` to an ISO image, but even in that case, Anaconda figures out what to do on its own and sets sources to local media (USB/CD/DVD/etc.), without any `cdrom` or `hd` option.

The autodetection is the default. So I would argue against using any one method as "default".